src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery {
+    private static final int MAX_NUMBER = 100;
+    private final Random random;
+
+    public Lottery(Random random) {
+        this.random = new Random();
+    }
+
+    public Ball getRandomBall() {
+        ColorSupplier colorSupplier = new ColorSupplier(random);
```

Comment:
Your colorSupplier is still created each time the method is called. Let's initialize it the same way as your 'random' object